### PR TITLE
added "ls" binary with basic functionality

### PR DESCRIPTION
**Pull request by uladzislaudd**

### CODE REVIEW

File: src/core/ls.c [hunk 1]
```diff
@@ -101,55 +103,104 @@ static STATUS ls_l_file(struct stat *s, struct ls_stat *ls)
     case __S_IFREG:  t = '-'; break;
     case __S_IFSOCK: t = '-'; break;
     }
```
Comment (dthmrphn) [inline src/core/ls.c]: default case missed?

File: src/core/ls.c [hunk 2]
```diff
@@ -101,55 +103,104 @@ static STATUS ls_l_file(struct stat *s, struct ls_stat *ls)
...
+        sprintf(ls->p, "\e[1;36m%s\e[0m -> %s", ls->n, ls1.p);
+        break;
+
+    case __S_IFREG:
+        sprintf(ls->p, "\e[0m%s\e[0m", ls->n);
+        break;
+
+    case __S_IFSOCK:
+        sprintf(ls->p, "\e[1;35m%s\e[0m", ls->n);
+        break;
```
Comment (dthmrphn) [inline src/core/ls.c]: default case?

File: src/core/ls.c [hunk 3]
```diff
@@ -101,55 +103,104 @@ static STATUS ls_l_file(struct stat *s, struct ls_stat *ls)
...
             goto exit;
         }
 
         ls_stat_lens(&(lss[i]), lens);
 
         i++;
     }
+    if (errno != 0) {
+        rv = errno;
+        goto exit;
```
Comment (dthmrphn) [inline src/core/ls.c]: do { } while(0); would be better

File: src/ls/ls.c [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    struct ls_stat ls1;
+
+    switch (s->st_mode & __S_IFMT) {
+    case __S_IFBLK:  t = '-'; break;
+    case __S_IFCHR:  t = 'c'; break;
+    case __S_IFDIR:  t = 'd'; break;
+    case __S_IFIFO:  t = '-'; break;
+    case __S_IFLNK:  t = 'l'; break;
+    case __S_IFREG:  t = '-'; break;
+    case __S_IFSOCK: t = '-'; break;
```
Comment (dthmrphn) [inline src/ls/ls.c]: Default case missed

File: src/ls/ls.c [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        sprintf(ls->p, "\e[1;36m%s\e[0m -> %s", ls->n, ls1.p);
+        break;
+
+    case __S_IFREG:
+        sprintf(ls->p, "\e[0m%s\e[0m", ls->n);
+        break;
+
+    case __S_IFSOCK:
+        sprintf(ls->p, "\e[1;35m%s\e[0m", ls->n);
+        break;
```
Comment (dthmrphn) [inline src/ls/ls.c]: Also no default case

File: src/core/ls.c (hunk 3 above)
Comment (uladzislaudd) [reply src/core/ls.c]: Good catch, but i gonna change this function in future.

**Review by dthmrphn — APPROVED** [review]
everything is okay

**Pull request merged**